### PR TITLE
feat: use shoutrrr for notifcations (apprise deprecated)

### PR DESCRIPTION
**Pull request by kmendell**
https://shoutrrr.nickfedor.com/v0.13.1/services/overview/

**Disclaimer** Greptiles Reviews use AI, make sure to check over its work

<details open><summary><h3>Greptile Summary</h3></summary>

This PR successfully migrates Arcane from Apprise API to Shoutrrr for direct notification support, eliminating the need for a separate Apprise container. The implementation adds comprehensive support for 19 notification providers (Discord, Email, Telegram, Slack, Gotify, ntfy, Pushbullet, Pushover, and many others via generic Shoutrrr URLs).

**Key improvements:**
- Direct Shoutrrr integration with proper webhook parsing for Discord/Slack
- Robust email validation with IDNA support and internationalized domain names
- Comprehensive email template rendering with HTML/multipart support
- TLS mode configuration (none/starttls/ssl) with certificate verification option
- Extensive test coverage including SMTP server testing and provider mocking
- Frontend validation with email normalization matching backend logic
- Deprecated Apprise endpoints properly marked
- Database migration adds provider-specific configuration storage

**Issues identified:**
- Webhook URL validation returns empty string instead of error (lines 32-36)
- Generic provider fallback doesn't validate URL presence (lines 38-42)
- Inefficient JSON marshaling in getURLFromConfig (line 178)
- Migration default empty string leaves existing rows with blank names (UX issue)
- Email toAddresses field lacks strict comma-separation validation on frontend

**Assessment:** The implementation is solid overall with good test coverage and proper error handling in most places. The identified issues are primarily around edge cases and UX polish rather than critical bugs. The codebase properly validates URL building at save time, preventing users from creating invalid notification settings that would only fail when sending.

</details>
<details open><summary><h3>Confidence Score: 4/5</h3></summary>

- This PR is safe to merge with minor quality-of-life improvements recommended for edge cases and UX
- Score reflects comprehensive implementation with proper error handling at critical points (URL validation during save), excellent test coverage covering SMTP, Discord, Slack, and multiple providers, and correct handling of email complexity (IDNA, multipart headers). Deductions for webhook validation edge cases, inefficient marshaling pattern, and migration UX issue that could confuse users with blank provider names. The identified issues are non-blocking as they don't cause runtime failures in normal usage.
- Migration default value for `name` column should be improved for better UX with existing notification settings. Webhook URL validation in shoutrrr_util.go should return explicit errors instead of silent failures.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| backend/internal/utils/notifications/shoutrrr_util.go | Shoutrrr URL builder with comprehensive provider support. Email validation with IDNA support is solid, and Discord/Slack webhook parsing is improved. However, webhook URL validation and error handling could be more explicit - webhook case (line 32-36) silently returns empty string if URL missing, and generic fallback (line 38-42) doesn't validate URL exists. |
| backend/internal/services/notification_service.go | Notification service implementation with Shoutrrr integration. BuildShoutrrrURL error handling is improved and properly validated at save time. Email template rendering is comprehensive with proper HTML/multipart handling. However, getURLFromConfig (line 178) performs inefficient JSON marshaling/unmarshaling of already-parsed config object. |
| backend/internal/huma/handlers/notifications.go | API handlers for notification settings with proper error handling and security checks. CreateOrUpdateNotificationSettings correctly passes config to service which validates URL building. Apprise endpoints are marked deprecated. |
| frontend/src/lib/components/sheets/add-notification-provider-sheet.svelte | Form component for notification provider configuration with client-side email validation. Email validation function with IDNA support and regex fallback is well-implemented. However, toAddresses field accepts free-text without enforcing comma separation format - users can enter invalid formats that only fail at backend. |
| backend/resources/migrations/postgres/029_add_name_to_notification_settings.up.sql | Migration adds name column to notification_settings table with empty string default. This leaves existing rows with blank names that display poorly in UI. Should use provider-based default or 'Notification #' || id format for better UX. |

</details>

</details>

**Context used:**

- Rule from `dashboard` - GoLang Best Practices

Follow idiomatic Go patterns and conventions
Handle errors explicitly, don’t ... ([source](https://app.greptile.com/review/custom-context?memory=214b40a8-9695-4738-986d-5949b5d65ff1))
- Rule from `dashboard` - JavaScript/TypeScript Best Practices

Use const/let instead of var for variable declarations
Prefer ... ([source](https://app.greptile.com/review/custom-context?memory=68c6ab5b-40de-4e3c-a803-95d60efb41a6))

### CODE REVIEW

**Comment by kmendell** [comment]
@greptileai

**Comment by kmendell** [comment]
@greptileai

**Comment by kmendell** [comment]
@greptileai

File: backend/resources/migrations/postgres/029_add_name_to_notification_settings.up.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE notification_settings ADD COLUMN name VARCHAR(255) DEFAULT '';
```
[the remark this answers is not in the document]
Comment (kmendell) [reply backend/resources/migrations/postgres/029_add_name_to_notification_settings.up.sql:1]: This is not a issue

**Comment by kmendell** [comment]
@greptileai

**Comment by Serph91P** [comment]
@kmendell When I try to add my ntfy I get an error that just says “Failed to add”. 

the log is this:
```
Jan 08 09:20:31.105 INF HTTP error: 500 internal server error request.method=POST request.host=arcane.domain.tld request.path=/api/environments/0/notifications/settings request.query="" request.params=map[id:0] request.route=/api/environments/:id/notifications/settings request.referer=https://arcane.domain.tld/settings/notifications request.length=110 response.latency=1.798292ms response.status=500 response.length=240
```

**Comment by kmendell** [comment]
@Serph91P whats is the url you used in the form?

**Comment by Serph91P** [comment]
> @Serph91P whats is the url you used in the form?

ntfy://ntfy.domain.tld/topic just as the documentation said.

**Comment by kmendell** [comment]
Moved this to the 1.13 milestone as i need to do more work.

**Comment by kmendell** [comment]
Am trying a second attempt at this.
